astropy/units/quantity_helper/converters.py

```diff
@@ -378,6 +378,9 @@ def check_output(output, unit, inputs, function=None):
         # if the output is used to store results of a function.
         return output.view(np.ndarray)
 
+    elif any(isinstance(output, type(inp)) for inp in inputs):
```

Comment:
Is this still necessary? I'd think that with the `try/except` we already return `NotImplemented`, which probably is the right call.

Reply:
I guess not, sorry about that! I don't remember why I had to add this but it definitely seems unnecessary with our current testing infrastructure.